Add support shared owners to transit gateway (#112)
C2Devel/terraform-provider-croccloud: website/docs/r/ec2_transit_gateway_project_access.md

---
subcategory: "Transit Gateway"
layout: "aws"
page_title: "aws_ec2_transit_gateway_project_access"
description: |-

>   Grants project access to transit gateway.

Reviewer: ```suggestion
  Grants project access to the transit gateway.
```

И в описании

---

# Resource: aws_ec2_transit_gateway_project_access

Grants project access to transit gateway.

## Example Usage

```terraform
resource "aws_iam_project" "example" {
  name = "tf-project"
}

resource "aws_ec2_transit_gateway" "example" {
  description = "tf example"

  tags = {
    Name = "tf-tgw"
  }
}

resource "aws_ec2_transit_gateway_project_access" "example" {
  transit_gateway_id = aws_ec2_transit_gateway.example.id
  project_id         = aws_iam_project.example.id
}
```

## Argument Reference

* `project_id` - (Required) The ID of the project (`project@customer`) that will be granted access to the transit gateway.
* `transit_gateway_id` - (Required) The ID of the transit gateway.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - The ID of the transit gateway project access, composed of `transit_gateway_id` and `project_id` separated by a forward slash (`/`). For example: `tgw-12345ABC/project@customer`.

Reviewer: ```suggestion
* `id` - `transit_gateway_id` and `project_id` separated by a forward slash (`/`). For example: `tgw-12345ABC/project@customer`.
```

Можно так

## Import

> In Terraform v1.5.0 and later, use an import block to import project access using the ID of the transit gateway and the ID of the project separated by a slash (`/`).

Reviewer: forward slash, раз уж выше так)

> In Terraform v1.5.0 and later, use an import block to import project access using the ID of the transit gateway and the ID of the project separated by a slash (`/`).

Reviewer: to import the resource, тк project access - немного абстрактно

For example:

```terraform
import {
  to = aws_ec2_transit_gateway_project_access.example
  id = "tgw-12345ABC/project@customer"
}
```

> This resource can be imported using the ID of the transit gateway and the ID of the project separated by a slash (`/`), e.g.,

Reviewer: Думаю, что эту фразу можно без потери смысла заменить нa: Or using `terraform import:`, чтобы было меньше дублирования.

Или как у aws:
Using `terraform import`, import the resource using the ID of the transit gateway and the ID of the project separated by a forward slash (`/`). For example:

---
subcategory: "Transit Gateway"
layout: "aws"
page_title: "aws_ec2_transit_gateway_project_access"
description: |-
  Grants project access to the transit gateway.
---

# Resource: aws_ec2_transit_gateway_project_access

Grants project access to the transit gateway.

~> **Note** This resource should not be used in conjunction with the `aws_ec2_transit_gateway` resource if the `shared_owners` argument is defined. For example, using `aws_ec2_transit_gateway` with `shared_owners` and `aws_ec2_transit_gateway_project_access` to manage sharing for the same Transit Gateway will cause a perpetual configuration drift. The `aws_ec2_transit_gateway` resource will continuously attempt to enforce its own sharing configuration, conflicting with the settings applied by `aws_ec2_transit_gateway_project_access`.

## Example Usage

```terraform
resource "aws_iam_project" "example" {
  name = "tf-project"
}

resource "aws_ec2_transit_gateway" "example" {
  description = "tf example"

  tags = {
    Name = "tf-tgw"
  }
}

resource "aws_ec2_transit_gateway_project_access" "example" {
  transit_gateway_id = aws_ec2_transit_gateway.example.id

>   account_id         = "${aws_iam_project.example.name}@${split(":", aws_iam_project.example.arn)[4]}"

Reviewer: Ухх, разбор арна на кусочки, жестко. Докинь может сюда коммент `# project@customer`, кажется так будет чуть понятнее, что скрывается за парсингом

---
subcategory: "Transit Gateway"
layout: "aws"
page_title: "aws_ec2_transit_gateway_project_access"
description: |-
  Grants project access to the transit gateway.
---

# Resource: aws_ec2_transit_gateway_project_access

Grants project access to the transit gateway.

> ~> **Note** Do not manage transit gateway sharing in more than one place. Use either:  

Reviewer 2: см. коммент выше (за исключением отступа)

Author: Исправил по аналогии

  \- [`aws_ec2_transit_gateway_project_access`](ec2_transit_gateway_project_access.md) resources.  
  &nbsp;  
  Using both for the same transit gateway will conflict, as both attempt to enforce sharing state.

## Example Usage

```terraform
resource "aws_iam_project" "example" {
  name = "tf-project"
}

resource "aws_ec2_transit_gateway" "example" {
  description = "tf example"

  tags = {
    Name = "tf-tgw"
  }
}

resource "aws_ec2_transit_gateway_project_access" "example" {
  transit_gateway_id = aws_ec2_transit_gateway.example.id

  # project@customer
  account_id = "${aws_iam_project.example.name}@${split(":", aws_iam_project.example.arn)[4]}"
}
```

## Argument Reference

* `account_id` - (Required) The ID of the project (`project@customer`) that will be granted access to the transit gateway.
* `transit_gateway_id` - (Required) The ID of the transit gateway.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - `transit_gateway_id` and `account_id` separated by a forward slash (`/`). For example: `tgw-12345ABC/project@customer`.

## Import

> In Terraform v1.5.0 and later, use an import block to import the resource using the ID of the transit gateway and the ID of the project separated by a forward slash (`/`).

Reviewer 2: По аналогии с исправлениями https://github.com/C2Devel/terraform-provider-rockitcloud/pull/111/files
```suggestion
In Terraform v1.5.0 and later, the resource can be imported by the ID of the transit gateway and the ID of the project separated by a forward slash (`/`) using the `import` block.
```

For example:

```terraform
import {
  to = aws_ec2_transit_gateway_project_access.example
  id = "tgw-12345ABC/project@customer"
}
```

> Using `terraform import`, import the resource using the ID of the transit gateway and the ID of the project separated by a forward slash (`/`). For example:

Reviewer 2: ```suggestion
In earlier Terraform versions, the resource can be imported by the ID of the transit gateway and the ID of the project separated by a forward slash (`/`) using `terraform import`, e.g.:
```
